client/launchpad/hooks/use-task-completed-notice.ts

```diff
@@ -0,0 +1,55 @@
+import page from '@automattic/calypso-router';
+import { updateLaunchpadSettings, useLaunchpad } from '@automattic/data-stores';
+import { useI18n } from '@wordpress/react-i18n';
+import { useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'calypso/state';
+import { successNotice } from 'calypso/state/notices/actions';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+interface TaskCompleteNoticeOptions {
+	taskSlug: string;
+	enabled?: boolean;
+	noticeId: string;
+	noticeText: string;
+	noticeDuration?: number;
+}
+
+export const useTaskCompletedNotice = ( {
+	taskSlug,
+	noticeId,
+	noticeText,
+	noticeDuration,
+	enabled = false,
```

Comment:
I like that it's off by default, otherwise it will be updating settings willy nilly. But I think we should make this a required argument, since off-by-default might leave the dev wondering why the hook isn't doing anything. Plus I think there should always be very specific logic for when the task gets completed or the notice gets shown. It seems we would only want the notice appearing when the user navigated directly from the launchpad, not when they were just organically navigating.